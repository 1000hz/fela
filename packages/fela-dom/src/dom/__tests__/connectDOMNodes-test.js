import connectDOMNodes from '../connectDOMNodes'

describe('Reflush style nodes', () => {
  it('should load and process style nodes', () => {
    const node1 = document.createElement('style')
    node1.setAttribute('data-fela-type', 'RULE')
    node1.type = 'text/css'
    node1.textContent = '.a{color:yellow}.a:hover{color:red}'

    const node2 = document.createElement('style')
    node2.setAttribute('data-fela-type', 'RULE')
    node2.type = 'text/css'
    node2.media = '(max-width: 800px)'
    node2.textContent = '.b{color:blue}'

    document.head.appendChild(node1)
    document.head.appendChild(node2)

    const renderer = {
      cache: {},
      getNextRuleIdentifier() {
        return true
      }
    }

    connectDOMNodes(renderer)

    expect(renderer.nodes).toEqual({
      RULE: node1,
      'RULE(max-width: 800px)': node2
    })
  })
})
