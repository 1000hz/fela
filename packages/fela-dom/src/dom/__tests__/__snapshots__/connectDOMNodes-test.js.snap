// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connecting DOM nodes should initialize nodes 1`] = `
Array [
  Array [
    "RULEfalse",
    "RULEtrue",
    "RULE(min-height: 800px)false",
    "RULE(min-height: 800px)true",
  ],
  "<html>

<head>
  <style data-fela-type=\\"RULE\\" type=\\"text/css\\">
    .a {
      color: red
    }
  </style>
  <style data-fela-type=\\"RULE\\" type=\\"text/css\\" data-fela-support=\\"true\\">
    @supports (display: flex) {
      .c {
        color: yellow
      }
    }
  </style>
  <style data-fela-type=\\"RULE\\" type=\\"text/css\\" media=\\"(min-height: 800px)\\">
    .b {
      color: blue
    }
  </style>
  <style data-fela-type=\\"RULE\\" type=\\"text/css\\" data-fela-support=\\"true\\" media=\\"(min-height: 800px)\\">
    @supports (display: flex) {
      .d {
        color: green
      }
    }
  </style>
</head>

<body></body>

</html>",
]
`;
